.github/actions/repo-gardening/src/tasks/check-description/index.js

```diff
@@ -0,0 +1,305 @@
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<string>} Promise resolving to info about the next release for that plugin.
+ */
+async function buildMilestoneInfo( octokit, owner, repo, number ) {
+	const plugins = await getPluginNames( octokit, owner, repo, number );
+	const ownerLogin = owner.login;
+	let pluginInfo;
+
+	// Get next valid milestone for each plugin.
+	for await ( const plugin of plugins ) {
+		const nextMilestone = await getNextValidMilestone( octokit, ownerLogin, repo, plugin );
+		debug( `check-description: Milestone found: ${ nextMilestone }` );
+
+		debug( `check-description: getting milestone info for ${ plugin }` );
+		const info = await getMilestoneDates( plugin, nextMilestone );
+
+		pluginInfo += info;
+	}
+
+	return pluginInfo;
+}
+
+/**
+ * Search for a previous comment from this task in our PR.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<number>} Promise resolving to boolean.
+ */
+async function getCheckComment( octokit, owner, repo, number ) {
+	let commentID = 0;
+
+	debug( `check-description: Looking for a previous comment from this task in our PR.` );
+
+	for await ( const response of octokit.paginate.iterator( octokit.issues.listComments, {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			if (
+				comment.user.login === 'github-actions[bot]' &&
+				comment.body.includes( '**Thank you for your PR!**' )
+			) {
+				commentID = comment.id;
+			}
+		} );
+	}
+
+	return commentID;
+}
+
+/**
+ * Checks the contents of a PR description.
+ *
+ * @param {WebhookPayloadPullRequest} payload - Pull request event payload.
+ * @param {GitHub}                    octokit - Initialized Octokit REST client.
+ */
+async function checkDescription( payload, octokit ) {
+	const { base, body, head, number } = payload.pull_request;
+	const { name: repo, owner } = payload.repository;
+
+	debug( `check-description: start building our comment` );
+
+	// We'll add any remarks we may have about the PR to that comment body.
+	let comment = `**Thank you for your PR!**
+
+When contributing to Jetpack, we have [a few suggestions](https://github.com/Automattic/jetpack/blob/master/.github/PULL_REQUEST_TEMPLATE.md) that can help us test and review your patch:<br>`;
+
+	// No PR is too small to include a description of why you made a change
+	comment += `
+- ${
+		body < 10 ? `:red_circle:` : `:white_check_mark:`
+	} Include a description of your PR changes.<br>`;
+
+	// Check all commits in PR.
+	const isDirty = await hasUnverifiedCommit( octokit, owner, repo, number );
+	comment += `
+- ${ isDirty ? `:red_circle:` : `:white_check_mark:` } All commits were linted before commit.<br>`;
```

Comment:
I note it's impossible to fix this one without a rebase, which would be kind of annoying. That matters now because the bot will remove the needs-review label.

Plus CI should now fail it anyway if there actually are any new lint errors. OTOH, CI doesn't catch things like prettier's changes.

Reply:
Good point. I've now updated the failure icon for this check so it doesn't trigger the label changes. 20c4c57d83226156a333fd34000005c34ffa5415 and 300d8bfc257a0c22efba0547ee2f46e697dbe561